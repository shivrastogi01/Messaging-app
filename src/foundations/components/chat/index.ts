export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
