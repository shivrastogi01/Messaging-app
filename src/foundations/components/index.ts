export * from "./presentation";
export * from "./layout";
