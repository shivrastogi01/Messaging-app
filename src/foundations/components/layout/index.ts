export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
