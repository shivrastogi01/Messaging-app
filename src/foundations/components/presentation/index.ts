export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
