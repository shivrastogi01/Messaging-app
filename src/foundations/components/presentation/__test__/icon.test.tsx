import React from "react";
import { Icon, Icons } from "foundations/components/presentation";
import renderWithTheme from "main/testing/render-with-theme";

describe("Icon", () => {
  test("add icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Add} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("back icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Back} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("cross icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Cross} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("emoji icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Emoji} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("giphy icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Giphy} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("leave icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Leave} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("logo icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Logo} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("people icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.People} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("presence icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Presence} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("search icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Search} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("send icon matches the snapshot", () => {
    const { container } = renderWithTheme(
      <Icon icon={Icons.Send} title="Title" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("clickable variant matches the snapshot", () => {
    const { container } = renderWithTheme(<Icon icon={Icons.Add} clickable />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
