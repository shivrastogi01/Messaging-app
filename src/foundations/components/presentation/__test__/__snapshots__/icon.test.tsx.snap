// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon add icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    add.svg
  </svg>
</object>
`;

exports[`Icon back icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    back.svg
  </svg>
</object>
`;

exports[`Icon clickable variant matches the snapshot 1`] = `
.c0 {
  cursor: pointer;
}

.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg>
    add.svg
  </svg>
</object>
`;

exports[`Icon cross icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    cross.svg
  </svg>
</object>
`;

exports[`Icon emoji icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    emoji.svg
  </svg>
</object>
`;

exports[`Icon giphy icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    giphy.svg
  </svg>
</object>
`;

exports[`Icon leave icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    leave.svg
  </svg>
</object>
`;

exports[`Icon logo icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    logo.svg
  </svg>
</object>
`;

exports[`Icon people icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    people.svg
  </svg>
</object>
`;

exports[`Icon presence icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    presence.svg
  </svg>
</object>
`;

exports[`Icon search icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    search.svg
  </svg>
</object>
`;

exports[`Icon send icon matches the snapshot 1`] = `
.c0 svg {
  display: block;
}

<object
  class="c0"
>
  <svg
    title="Title"
  >
    send.svg
  </svg>
</object>
`;
